source/common/buffer/buffer_impl.cc

```diff
@@ -77,6 +85,7 @@ void OwnedImpl::prepend(Instance& data) {
     uint64_t slice_size = other.slices_.back().dataSize();
     length_ += slice_size;
     slices_.emplace_front(std::move(other.slices_.back()));
+    slices_.front().maybeChargeAccount(account_);
```

Comment:
Does this need to be called anytime a slice is added to `slices_`? If so, would it make sense to wrap `slices_` in a class that enforces calling `maybeChargeAccount()` when slices are added? Or is that more trouble than it's worth?

Reply:
It's not always called when added slices -- for example slices that are `BufferFragments`. It seems like it could be more trouble than it's worth.